### PR TITLE
[Merged by Bors] - feat: Set explicit resources on all containers

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of: https://github.com/stackabletech/issues/issues/394

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/druid/pages/usage-guide/resources-and-storage.adoc
Comment (sbernauer) [inline docs/modules/druid/pages/usage-guide/resources-and-storage.adoc:14]: We include it just above already
```suggestion
```

File: docs/modules/druid/pages/usage-guide/resources-and-storage.adoc
Comment (sbernauer) [inline docs/modules/druid/pages/usage-guide/resources-and-storage.adoc:51]: In the other operators we have used 1500Mi, would leave it up to you to decide
Applies to all the code below as well

File: tests/templates/kuttl/smoke/30-install-zk.yaml.j2
Comment (sbernauer) [inline tests/templates/kuttl/smoke/30-install-zk.yaml.j2:19]: My recommendation would be to remove this lines, so that we implicitly test the default zk resources (are they enough?)

File: tests/templates/kuttl/smoke/40-install-hdfs.yaml.j2
Comment (sbernauer) [inline tests/templates/kuttl/smoke/40-install-hdfs.yaml.j2:26]: 2x times below as well:

My recommendation would be to remove this lines, so that we implicitly test the default hdfs resources

File: tests/templates/kuttl/smoke/50-install-druid.yaml.j2
Comment (sbernauer) [inline tests/templates/kuttl/smoke/50-install-druid.yaml.j2:50]: Intentional? I hope we don't need this change or?

**Review by sbernauer — COMMENTED** [review]
Only minor things

File: tests/templates/kuttl/smoke/50-install-druid.yaml.j2
Comment (razvan) [reply tests/templates/kuttl/smoke/50-install-druid.yaml.j2]: done

File: tests/templates/kuttl/smoke/40-install-hdfs.yaml.j2
Comment (razvan) [reply tests/templates/kuttl/smoke/40-install-hdfs.yaml.j2]: done

File: tests/templates/kuttl/smoke/30-install-zk.yaml.j2
Comment (razvan) [reply tests/templates/kuttl/smoke/30-install-zk.yaml.j2]: done

**Review by sbernauer — APPROVED** [review]
LGTM when tests pass

**Comment by razvan** [comment]
Latest tests pass

```
--- PASS: kuttl (2419.95s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/hdfs-deep-storage_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (445.72s)
        --- PASS: kuttl/harness/logging_druid-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (520.83s)
        --- PASS: kuttl/harness/resources_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev (198.70s)
        --- PASS: kuttl/harness/ldap-authentication_druid-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_opa-0.45.0-stackable0.0.0-dev_hadoop-latest-3.3.4-stackable0.0.0-dev_ldap-use-tls-true_ldap-no-bind-credentials-false_openshift-true (352.87s)
        --- PASS: kuttl/harness/s3-deep-storage_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_s3-use-tls-true (314.49s)
        --- PASS: kuttl/harness/ingestion-s3-ext_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (396.33s)
        --- PASS: kuttl/harness/authorizer_druid-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_opa-0.45.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (376.75s)
        --- PASS: kuttl/harness/cluster-operation_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-latest-3.3.4-stackable0.0.0-dev (426.07s)
        --- PASS: kuttl/harness/tls_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_tls-mode-internal-and-server-tls-and-tls-client-auth_openshift-true (320.70s)
        --- PASS: kuttl/harness/orphans_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (405.80s)
        --- PASS: kuttl/harness/smoke_druid-26.0.0-stackable0.0.0-dev_zookeeper-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev_openshift-true (347.24s)
        --- PASS: kuttl/harness/ingestion-no-s3-ext_druid-latest-26.0.0-stackable0.0.0-dev_zookeeper-latest-3.8.0-stackable0.0.0-dev_hadoop-3.3.4-stackable0.0.0-dev (381.13s)
```

**Comment by razvan** [comment]
bors merge

File: docs/modules/druid/pages/usage-guide/resources-and-storage.adoc
Comment (sbernauer) [inline docs/modules/druid/pages/usage-guide/resources-and-storage.adoc:7]:
```suggestion
== Resource Requests
```

**Review by sbernauer — CHANGES_REQUESTED** [review]

**Comment by sbernauer** [comment]
bors cancel

**Review by sbernauer — APPROVED** [review]

**Comment by sbernauer** [comment]
Sorry, missed that change

**Comment by razvan** [comment]
bors merge

**Comment by razvan** [comment]
bors cancel

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**
